lib/src/widgets/board_editor.dart

```diff
@@ -0,0 +1,274 @@
+import 'package:chessground/src/board_geometry.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+
+import 'piece.dart';
+import 'positioned_square.dart';
+import '../models.dart';
+import '../fen.dart';
+import '../board_settings.dart';
+
+class BoardEditor extends StatefulWidget with BoardGeometry {
+  const BoardEditor({
+    super.key,
+    required this.size,
+    required this.orientation,
+    required this.initialFen,
+    this.settings = const BoardSettings(),
```

Comment:
I know this is still WIP, but we should have a dedicated settings class for the editor.

Reply:
Yeah good point. Do you think it makes sense to factor out the common values related to appearance/theming (colorScheme, pieceAssets, borderRadius, boxShadow, ...) into a common `VisualBoardSettings` class? Or should we just duplicate them in `BoardEditorSettings` instead?